drone-ci/drone/resources.tf

```diff
@@ -1,87 +1,38 @@
-
-
-provider "aws" {
-  access_key = "${var.aws_access_key}"
-  secret_key = "${var.aws_secret_key}"
-  region = "${var.aws_region}"
-}
-
-resource "aws_key_pair" "user" {
-  key_name = "${var.keypair_name}"
-  public_key = "${var.keypair_key}"
-}
-
-resource "aws_security_group" "drone" {
-  name = "${var.aws_security_group_name}"
-  description = "${var.aws_security_group_name}"
-  vpc_id = "${var.drone_vpc_id}"
-  ingress {
-    from_port = 22
-    to_port = 22
-    protocol = "tcp"
-    cidr_blocks = [
-     "0.0.0.0/0"
-    ]
-  }
-  ingress {
-    from_port = 8000
-    to_port = 8000
-    protocol = "tcp"
-    cidr_blocks = [
-     "0.0.0.0/0"
-    ]
-  }
-  egress {
-    from_port = 0
-    to_port = 0
-    protocol = "-1"
-    cidr_blocks = [
-      "0.0.0.0/0"
-    ]
-  }
-
-  tags {
-    Name = "${var.aws_security_group_name}"
-  }
-}
-
-resource "aws_ebs_volume" "drone-volume" {
-  availability_zone = "${data.aws_availability_zones.available-zones.names[0]}"
-  type = "${var.aws_volume_type}"
-  size = "${var.drone_volume_size}"
+resource "aws_ebs_volume" "volume" {
+  availability_zone = "${var.aws_zone}"
+  type = "${var.volume_type}"
+  size = "${var.volume_size}"
   tags {
-    Name = "${var.aws_volume_tag}"
+    Name = "${var.volume_tag}"
   }
 }
 
 resource "aws_instance" "drone" {
-  key_name = "${var.keypair_name}"
+  key_name = "${var.ssh_key_name}"
   ami = "${var.aws_base_ami}"
-  instance_type = "${var.aws_instance_type}"
-  availability_zone = "${data.aws_availability_zones.available-zones.names[0]}"
-  subnet_id = "${var.drone_subnet_id}"
-  vpc_security_group_ids = [
-    "${aws_security_group.drone.id}"
-  ]
-  private_ip = "${var.drone_private_ip}"
+  instance_type = "${var.instance_type}"
+  availability_zone = "${var.aws_zone}"
+  subnet_id = "${var.aws_subnet}"
+  vpc_security_group_ids = ["${var.aws_security_groups}"]
```

Comment:
yes I do believe this is the way to go. As i said in the PR summary having security groups defined in the module doesn't allow us to apply custom rules but just the ones pre defined by the module.